kratos/utilities/normal_calculation_utils.h

```diff
@@ -424,18 +474,40 @@ class KRATOS_API(KRATOS_CORE) NormalCalculationUtils
     ///@name Private Operations
     ///@{
 
-    //this function adds the Contribution of one of the geometries
-    //to the corresponding nodes
+    /**
+     * @brief It initializes the normal in the entites and in all the nodes
+     * @param rModelPart The model part to compute
+     * @tparam TEntity The entity type considered
+     */
+    template<class TEntity>
+    void InitializeNormals(ModelPart& rModelPart);
+
+    /**
+     * @brief It computes the unit normals from the area normals
+     * @param rModelPart The model part to compute
+     */
+    void ComputeUnitNormalsFromAreaNormals(ModelPart& rModelPart);
+
+    /**
+     * @brief This function adds the Contribution of one of the geometries to the corresponding nodes
+     * @param rAn Area normal
+     */
     static void CalculateNormal2D(
-        ConditionsArrayType::iterator it,
-        array_1d<double,3>& An
+        ConditionsArrayType::iterator itCond,
```

Comment:
since this is a big refactor can you please directly use a ´const Condition&` here?

Or if this is still needed for backward compatibility make a second function and deprecate this one

Reply:
i agree with Vicrntr on this. initializing from an impodsible number obliges to read

Reply:
Cannot be const as it sets the NORMAL